Select component with theme variables

richardhjtan opened the pull request:
https://github.com/user-attachments/assets/37cdab07-09ff-49b9-94b8-266ed407cf41

richardhjtan committed:
Select component with theme variables

burieberry commented on packages/boxel-ui/addon/src/components/select/index.gts:
```diff
@@ -20,103 +23,608 @@ interface Signature<ItemT = any> {
+      .variant-primary {
+        --select-background-color: var(
+          --boxel-select-background-color,
+          var(--primary, var(--boxel-600))
+        );
+        --select-border-color: var(
+          --boxel-select-border-color,
+          var(--primary, var(--boxel-600))
```
fallback should be `--boxel-highlight` for `--primary`

burieberry commented on packages/boxel-ui/addon/src/components/select/index.gts:
```diff
@@ -20,103 +23,608 @@ interface Signature<ItemT = any> {
+        );
+        --select-text-color: var(
+          --boxel-select-text-color,
+          var(--primary-foreground, var(--boxel-light))
```
boxel fallback should be `--boxel-dark` for `--primary-foreground`

burieberry commented on packages/boxel-ui/addon/src/components/select/index.gts:
```diff
@@ -20,103 +23,608 @@ interface Signature<ItemT = any> {
+      .variant-destructive {
+        --select-background-color: var(
+          --boxel-select-background-color,
+          var(--destructive, var(--boxel-600))
+        );
+        --select-border-color: var(
+          --boxel-select-border-color,
+          var(--destructive, var(--boxel-600))
```
fallback is `--boxel-danger` for `--destructive`

burieberry commented on packages/boxel-ui/addon/src/components/select/index.gts:
```diff
@@ -20,103 +23,608 @@ interface Signature<ItemT = any> {
+        );
+        --select-text-color: var(
+          --boxel-select-text-color,
+          var(--destructive-foreground, var(--boxel-light))
```
fallback is `--boxel-light` for `--destructive-foreground`

burieberry reviewed (changes requested):
Thanks for working on this tricky one. A couple of things:

1- Make sure the fallbacks for the `default` are matching the current Select component, including the placeholder color and the disabled state, or even better, if they are matching gardella's design. Looks like we're missing the checkmark icon for the selected item and the trigger border should turn black on hover and active. 

Here's the existing:
<img width="858" height="276" alt="existing" src="https://github.com/user-attachments/assets/eaf5c5e8-3fb6-4f36-8383-1dc17c2e290f" />

Here's the new one, which is not matching:
<img width="847" height="298" alt="new" src="https://github.com/user-attachments/assets/d5745836-9996-40e0-b270-b89c1759d332" />

Gardella design:
<img width="1117" height="369" alt="dropdown-design" src="https://github.com/user-attachments/assets/4db18ad1-9b00-4775-84cb-c9cac2bc8ea8" />

2- Bonus fix: It would be great if the usage doc didn't display [object Object]. Maybe convert it to text, or create a ticket for it to be handled at a later time.

richardhjtan committed:
Select feedback

richardhjtan committed:
Merge branch 'main' into CS-9307-select-with-theme-variables

richardhjtan committed:
Fix test and glint

richardhjtan commented:
Thanks for the feedback @burieberry , addressed the feedback and followed Gardella's design as possible
<img width="1085" height="371" alt="Screenshot 2025-08-25 at 4 22 54 PM" src="https://github.com/user-attachments/assets/483f0d71-20d7-4c5f-b39b-fe459e7e56c3" />

richardhjtan committed:
Focus and scroll selected item

burieberry reviewed (commented):
The `default` variant's default styles look good now. However I noticed that the  keyboard accessibility of this component is broken and seems to be related to our styling. Ember-power-select by itself (https://ember-power-select.com/docs/how-to-use-it) has the accessibility features.

burieberry commented on packages/boxel-ui/addon/src/components/select/trigger.gts:
```diff
@@ -57,6 +57,8 @@ export class BoxelTriggerWrapper extends Component<TriggerSignature> {
         width: 100%;
         gap: var(--boxel-sp-xxxs);
         padding: var(--boxel-sp-xs);
+        font: var(--boxel-font-sm);
```
whenever this shorthand `font` property is used, need to add `font-family: inherit` after it

richardhjtan replied:
Ah missed that out, added!

richardhjtan committed:
Select component accessibility

richardhjtan commented:
Fixed the accessibility which should be similar to `ember-power-select` now. Have to clear the cache when seeing preview https://cs-9307-select-with-theme-variables.boxel-ui-preview.stack.cards/?s=Components&ss=%3CSelect%3E
